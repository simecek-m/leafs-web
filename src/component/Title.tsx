import { FC } from "react";

interface TitleProps {
  children: string;
  className?: string;
}

const Title: FC<TitleProps> = ({ children, className }) => {
  return (
    <h1
      className={`${className} inline-block bg-gradient-to-br from-primary-400 to-primary-900 bg-clip-text font-heading text-5xl font-black text-transparent`}
    >
      {children}
    </h1>
  );
};

export default Title;
