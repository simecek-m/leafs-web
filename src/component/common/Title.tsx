import { FC } from "react";

interface TitleProps {
  children: string;
}

export const Title: FC<TitleProps> = ({ children }) => {
  return (
    <h1 className="inline-block font-heading text-5xl font-black text-brand-light dark:text-brand-dark">
      {children}
    </h1>
  );
};
