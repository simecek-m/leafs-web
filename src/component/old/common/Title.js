import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPencilAlt } from "@fortawesome/free-solid-svg-icons";

function Title({ text = "Title", icon = faPencilAlt }) {
  const iconComponent = icon ? <FontAwesomeIcon icon={icon} /> : null;
  return (
    <h1 className="title">
      {iconComponent} {text}
    </h1>
  );
}

export default Title;
