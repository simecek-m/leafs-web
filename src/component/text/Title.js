import "component/text/Title.sass";

export default function Title({ children }) {
  return <h1>{children}</h1>;
}
