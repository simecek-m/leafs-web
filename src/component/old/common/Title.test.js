import React from "react";
import Title from "component/old/common/Title";
import { shallow } from "enzyme";
import { faDatabase } from "@fortawesome/free-solid-svg-icons";

const TEST_CUSTOM_TITLE_TEXT = "Profile";
const TEST_CUSTOM_TITLE_ICON = faDatabase;

describe("Title component", () => {
  test("should render default Title", () => {
    const wrapper = shallow(<Title />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should render custom Title", () => {
    const wrapper = shallow(
      <Title text={TEST_CUSTOM_TITLE_TEXT} icon={TEST_CUSTOM_TITLE_ICON} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("should render custom Title without icon", () => {
    const wrapper = shallow(
      <Title text={TEST_CUSTOM_TITLE_TEXT} icon={null} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
