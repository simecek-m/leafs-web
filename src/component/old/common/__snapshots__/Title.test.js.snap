// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should render custom Title 1`] = `
<h1
  className="title"
>
  <FontAwesomeIcon
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f1c0",
          "M448 73.143v45.714C448 159.143 347.667 192 224 192S0 159.143 0 118.857V73.143C0 32.857 100.333 0 224 0s224 32.857 224 73.143zM448 176v102.857C448 319.143 347.667 352 224 352S0 319.143 0 278.857V176c48.125 33.143 136.208 48.572 224 48.572S399.874 209.143 448 176zm0 160v102.857C448 479.143 347.667 512 224 512S0 479.143 0 438.857V336c48.125 33.143 136.208 48.572 224 48.572S399.874 369.143 448 336z",
        ],
        "iconName": "database",
        "prefix": "fas",
      }
    }
  />
   
  Profile
</h1>
`;

exports[`Title component should render custom Title without icon 1`] = `
<h1
  className="title"
>
   
  Profile
</h1>
`;

exports[`Title component should render default Title 1`] = `
<h1
  className="title"
>
  <FontAwesomeIcon
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f303",
          "M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z",
        ],
        "iconName": "pencil-alt",
        "prefix": "fas",
      }
    }
  />
   
  Title
</h1>
`;
